vmf/scripts/mods/vmf/modules/ui/options/vmf_options_view.lua

```diff
@@ -935,61 +996,45 @@ local function create_header_widget(widget_definition, scenegraph_id)
 
+            -- Events & callbacks
             if is_interactable then
 
               if content.tooltip_text then
                 style.tooltip_text.cursor_offset = content.callback_fit_tooltip_to_the_screen(content, style.tooltip_text, ui_renderer)
               end
 
-              if content.highlight_hotspot.on_release and not content.checkbox_hotspot.on_release and not content.fav_icon_hotspot.on_release
-                and not content.fav_arrow_up_hotspot.on_release and not content.fav_arrow_down_hotspot.on_release then
-
-                content.callback_hide_sub_widgets(content)
-              end
-
-              if content.fav_icon_hotspot.on_release and not content.fav_arrow_up_hotspot.on_release and not content.fav_arrow_down_hotspot.on_release then
-                content.callback_favorite(content)
-              end
-
               if content.fav_arrow_up_hotspot.on_release then
                 content.callback_move_favorite(content, true)
-              end
-
-              if content.fav_arrow_down_hotspot.on_release then
+              elseif content.fav_arrow_down_hotspot.on_release then
                 content.callback_move_favorite(content, false)
-              end
-
-              if content.checkbox_hotspot.on_release then
-
-                if content.is_widget_collapsed then
-                  content.callback_hide_sub_widgets(content)
-                end
-
-                local mod_name       = content.mod_name
+              elseif content.fav_icon_hotspot.on_release then
+                content.callback_favorite(content)
+              elseif content.checkbox_hotspot.on_release then
                 local is_mod_enabled = not content.is_checkbox_checked
+                local mod_name = content.mod_name
 
                 content.is_checkbox_checked = is_mod_enabled
-
                 content.callback_mod_state_changed(mod_name, is_mod_enabled)
+              elseif content.highlight_hotspot.on_release then
+                content.callback_hide_sub_widgets(content)
               end
             end
 
-            content.fav_icon_texture = content.is_favorited and "header_fav_icon_lit" or "header_fav_icon"
-            --content.checkbox_texture = content.is_checkbox_checked and "checkbox_checked" or "checkbox_unchecked"
+            if content.is_favorited then
+              style.fav_icon.color = is_interactable and content.fav_icon_hotspot.is_hover and { 200, 255, 255, 255 } or { 255, 255, 255, 255 }
+            else
+              style.fav_icon.color = is_interactable and content.fav_icon_hotspot.is_hover and { 255, 255, 255, 255 } or { 125, 255, 255, 255 }
+            end
+
+            local is_disabled = content.is_checkbox_visible and not content.is_checkbox_checked
+            style.text.text_color = is_disabled and { 255, 100, 100, 100 } or { 255, 255, 255, 255 }
+            style.background.color = is_disabled and { 255, 175, 100, 100 } or { 255, 255, 225, 225 }
```

Comment:
A new table is being allocated for the color **every tick**! A possible alternative is storing the colors in the style table (so they're easier to change) and only change the reference:

```suggestion
            style.background.color = is_disabled and style.background.color_disabled or style.background.color_enabled
```

I realize the creation of tables is being carried over from the old version, but given that there's some refactoring going on I think it's a good time to bring this up for discussion.